src/syntax.ml

```diff
@@ -170,12 +175,14 @@ let renamed_in t ver ~to_ =
     Error.renamed_in loc t ver ~what ~to_
   end
 
-let since t ver =
+let since ?(fatal=true) t ver =
   let open Version.Infix in
   get_exn t >>= fun current_ver ->
   if current_ver >= ver then
     return ()
-  else begin
-    desc () >>= fun (loc, what) ->
-    Error.since loc t ver ~what
-  end
+  else
+    desc () >>= function
+    | (loc, what) when fatal -> Error.since loc t ver ~what
+    | (loc, what) ->
+      Errors.warn loc "%s" @@ Error_msg.since t ver ~what;
```

Comment:
Hmm, you can't pass `fatal` to `Error.since` because of types, right? I wonder if you could instead pass the continuation to it (`Errors.warn loc` or `Errors.fail loc`).

Reply:
(you'll probably need to use ksprintf)

Reply:
Will it really look any better?

If I understand correctly, you're suggesting to either pass `Errors.fail loc` or `fun s -> Errors.warn loc s; return ()` is that right?

If not I don't think I get what you're suggesting.

Reply:
Hmm I expected this to be shorter. Probably not much of an improvement then.

Reply:
Also note that `Syntax.Error.since` is exposed so when trying that out I had to make the continuation argument optional, meaning I couldn't remove the `loc` argument because it's required for the default continuation `Errors.fail loc "%s"`.

A solution might be to add some kind of `non_fatal_default: 'a` argument which would mean `fatal:false` and would specify what to return then.

Reply:
Just pushed a commit with the solution described above, let me know what you think!